source/slang/hlsl.meta.slang

```diff
@@ -3292,6 +3301,25 @@ bool WaveIsFirstLane()
     return WaveMaskIsFirstLane(WaveGetActiveMask());
 }
 
+// It's useful to have a wave uint4 version of countbits, because some wave functions return uint4.
+// This implementation tries to limit the amount of work required by the actual lane count.
+uint _WaveCountBits(uint4 value)
+{
+    // Assume since WaveGetLaneCount should be known at compile time, the branches will boil
+    // away
+    uint waveLaneCount = WaveGetLaneCount();
+    
+    switch (waveLaneCount / 32)
```

Comment:
I almost asked if this should be `((waveLaneCount + 31) / 32)`, but then I saw that your `case`s are set up to match this style.

Given this approach, though, it seems like you *do* want to change to `((waveLaneCount - 1) / 32)`, because otherwise the case where you have a 32-lane wave would map to `case 1` and add up the bits from both `.x` and `.y`.

There is also the argument to be made that if the downstream compiler can optimize for `WaveGetLaneCount()` returning a constant, then they can probably also optimize for `WaveActiveBallot()` and similar operations always returning zero in the `.y`, `.z`, and `.w` components...

Reply:
Good catch, fixed in PR.